### PR TITLE
WIP: Add metrics component (micrometer migration)

### PR DESCRIPTION
**Pull request by clement26695**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,8 @@
     "reactstrap": "^5.0.0"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^1.2.8",
```
Comment (deepu105) [inline package.json]: fix versions plz

Comment (deepu105) [reply package.json]: I don't like bundling font awesome with this lib, which icons are used?

File: src/component/metrics/jvmThreads.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  renderModal = () => <ThreadsModal handleClose={this.handleClose} showModal={this.state.showModal} threadDump={this.props.jvmThreads} />;
+
+  render() {
+    const { wholeNumberFormat } = this.props;
+    const { threadStats } = this.state;
+    return (
+      <div>
+        <b>Threads</b> (Total: {threadStats.threadDumpAll}){' '}
+        <Button color="link" className="hand" onClick={this.openModal}>
+          <FontAwesomeIcon icon="eye" />
```
Comment (deepu105) [inline src/component/metrics/jvmThreads.tsx]: use a normal small button instead of the icon and remove the font awesome dependency here

**Comment by deepu105** [comment]
Instead of redoing this for each technology, we should consider building this as a web component and using it in React, Angular and Vue. May be using something like stencil https://stenciljs.com/docs/my-first-component/

**Comment by clement26695** [comment]
It is indeed painful to write everything two or three times in each framework. I'll take a look at stencil to see how it works.

However, it looks like the work is almost done on these screens (except for Vue). Should we wait until I rewrite everything with stencil ? I don't know how long it is to get familiar with these tools

**Comment by deepu105** [comment]
I would prefer to give that a go before merging these into the libs

Thanks & Regards,
Deepu

**Comment by deepu105** [comment]
Can you try to do a POC with stencil to see if its feasible? you dont have
to replace everything may be just make a simple inner componenent and use
it along with this PR to see if it works

Thanks & Regards,
Deepu

**Comment by clement26695** [comment]
I am trying to do it. I created a small stencil library and tried to import a component in a jhipster app based on angular and another based on react. I managed to do it in the angular app but didn't managed to do it in the react one yet : following the stencil documentation, I have the error `Property 'jhi-jvm-memory-component' does not exist on type 'JSX.IntrinsicElements`

I created two demo repo : 

- one for the stencil lib : https://github.com/clement26695/demo-stencil-jhipster
- another for the angular app : https://github.com/clement26695/demo-stencil-jhipster-angular (the imported stencil component is in the metrics page)

**Comment by deepu105** [comment]
Ok i'll take a look at the React one

**Comment by deepu105** [comment]
Its a lint error actually, since we have configured tslint to allow only known elements and attributes

**Pull request merged**